test/extensions/tracers/skywalking/tracer_test.cc

```diff
@@ -0,0 +1,195 @@
+#include "extensions/tracers/skywalking/skywalking_client_config.h"
+#include "extensions/tracers/skywalking/tracer.h"
+
+#include "test/extensions/tracers/skywalking/skywalking_test_helper.h"
+#include "test/mocks/common.h"
+#include "test/mocks/server/tracer_factory_context.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/mocks/tracing/mocks.h"
+#include "test/mocks/upstream/cluster_manager.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace testing;
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace SkyWalking {
+
+class TracerTest : public testing::Test {
+public:
+  void setupTracer(const std::string& yaml_string) {
+    EXPECT_CALL(mock_dispatcher_, createTimer_(_)).WillOnce(Invoke([this](Event::TimerCb timer_cb) {
+      timer_cb_ = timer_cb;
+      return timer_;
+    }));
+
+    timer_ = new NiceMock<Event::MockTimer>();
```

Comment:
you can just instantiate as class normally? i.e. no need to be a pointer.

Reply:
The callback function takes ownership of the MockTimer object pointed to by the timer_ and then destroys it. It is necessary to use `new` to create a MockTimer object to ensure that it will not be destructed repeatedly.

```
    EXPECT_CALL(mock_dispatcher_, createTimer_(_)).WillOnce(Invoke([this](Event::TimerCb timer_cb) {
      timer_cb_ = timer_cb;
      return timer_;
    }));
```